File: uploadservice-okhttp/src/main/java/net/gotev/uploadservice/okhttp/OkHttpBodyWriter.java
package net.gotev.uploadservice.okhttp;

import net.gotev.uploadservice.http.BodyWriter;

import java.io.IOException;

import okio.BufferedSink;

/**
 * @author Aleksandar Gotev
 */

public class OkHttpBodyWriter extends BodyWriter {

    private BufferedSink mSink;

    protected OkHttpBodyWriter(BufferedSink sink) {
        mSink = sink;
    }

    @Override
    public void write(byte[] bytes) throws IOException {
        mSink.write(bytes);
    }

    @Override
    public void write(byte[] bytes, int lengthToWriteFromStart) throws IOException {
        mSink.write(bytes, 0, lengthToWriteFromStart);
    }

    @Override
    public void flush() throws IOException {
        mSink.flush();
    }
}

File: uploadservice-okhttp/src/main/java/net/gotev/uploadservice/okhttp/OkHttpBodyWriter.kt
package net.gotev.uploadservice.okhttp

import java.io.IOException
import net.gotev.uploadservice.network.BodyWriter
import okio.BufferedSink

/**
 * @author Aleksandar Gotev
 */

class OkHttpBodyWriter(private val sink: BufferedSink, listener: OnStreamWriteListener) :
    BodyWriter(listener) {
    @Throws(IOException::class)
    override fun internalWrite(bytes: ByteArray) {
        sink.write(bytes)
    }

    @Throws(IOException::class)
    override fun internalWrite(bytes: ByteArray, lengthToWriteFromStart: Int) {
        sink.write(bytes, 0, lengthToWriteFromStart)
    }

    @Throws(IOException::class)
    override fun flush() {
        sink.flush()
    }

    @Throws(IOException::class)
    override fun close() {
        sink.close()
    }
}
